### PR TITLE
fix STRICT_DNS_CLUSTER ResolveTarget cannot stop

### PR DESCRIPTION
**Pull request by bincherry**
### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/upstream/cluster/strict_dns_cluster.go [hunk 1]
```diff
@@ -251,6 +251,7 @@ func (rt *ResolveTarget) StartResolve() {
 		case <-rt.stop:
 			rt.resolveTimeout.Stop()
 			rt.resolveTimer.Stop()
+			return
```
Comment (wangfakang) [inline pkg/upstream/cluster/strict_dns_cluster.go:254]: Thank you for fixing it.
The following [case <-rt.stop](https://github.com/mosn/mosn/pull/1894/files#diff-c36f7a5b9260c86ee2d12759eb6b391caf06b96ed271f4fe52b285ce32ed90e1R257) can also be returned directly, thus reducing a select.

File: pkg/upstream/cluster/strict_dns_cluster.go [hunk 2]
```diff
@@ -251,6 +251,7 @@ func (rt *ResolveTarget) StartResolve() {
 		case <-rt.stop:
 			rt.resolveTimeout.Stop()
 			rt.resolveTimer.Stop()
```
Comment (doujiang24) [inline pkg/upstream/cluster/strict_dns_cluster.go:253]: better to remove these two lines? since they are already in the defer function.

Comment (doujiang24) [reply pkg/upstream/cluster/strict_dns_cluster.go:253]: Also, better to add the debug log here? just like the next one.

**Review by doujiang24 — COMMENTED** [review]
Nice Catch.

**Review by doujiang24 — APPROVED** [review]

**Review by wangfakang — APPROVED** [review]
LGTM

**Comment by bincherry** [comment]
Thanks for your help.

**Pull request merged**